rust/scx_utils/src/compat.rs

```diff
@@ -145,11 +149,28 @@ pub fn struct_has_field(type_name: &str, field: &str) -> Result<bool> {
 }
 
 pub fn ksym_exists(ksym: &str) -> Result<bool> {
-    let btf: &btf = *VMLINUX_BTF;
+    let file = std::fs::File::open("/proc/kallsyms")?;
+    let reader = std::io::BufReader::new(file);
+    let mut syms_present: HashSet<String> = HashSet::new();
 
-    let ksym_name = CString::new(ksym).unwrap();
-    let tid = unsafe { btf__find_by_name(btf, ksym_name.as_ptr()) };
-    Ok(tid >= 0)
+    for line in reader.lines() {
+        for sym in line.unwrap().split_whitespace() {
+            syms_present.insert(sym.into());
+        }
+    }
+
+    Ok(syms_present.contains(ksym))
```

Comment:
There's no reason to create a set here. Return true from the for loop if the string matches, else return false at the end of the function.

Reply:
The premise of this assertion is us knowing that checking every entry in this split list with duplicates is cheaper than inserting this split list with duplicates into a set and checking if that set contains the passed in string. Do we know this?